### DISCUSSION TITLE
Show default values in usage

### DISCUSSION THREAD

**Discussion by gmagnu** [ToToTec/CmdOption#19] [Ideas]
It would be nice to show the default values with the usage information.

**Comment by lefou** [comment]
What exactly do you have in mind when you say "default value"?

CmdOption itself has no notion of a default value. It especially does not set a default value in case the user provided no option. All it does is to **not** set values for **not** given options.

**Comment by lefou** [comment]
There are some posibilities, e.g.:

* Provide a textual default value via annottion (`@CmdOption(..., defaultValue="1")
* Provide a default value getter method (`public int defaultLogLevel() { return 1; }`)
* Provide some kind of strategie to find default values, e.g. search for getter methods and fields with same name
* Provide a annotation value, which is evaluated lika a script, so that it is possible to provide the default value directly or else invoke some code to compute it (@CmdOption(..., defaultValue="getLogLevel()"`)

**Comment by gmagnu** [comment]
Consider the following option definitions:

`@CmdOption(names = {"--limit", "-l"}, args = {"<limit>"}, description = "Limit output. Defaults to 1000.")`
`protected int limit = 1000;`
`@CmdOption(names = {"--offset", "-o"}, args = {"<offset>"}, description = "Offset into output. Defaults to 0.")`
`protected int offset = 0;`

Here I have two optional options, limit and offset, where limit defaults to 1000 and offset to 0 (which are replaced if you provide the option).   As you see I am putting the defaults in the description and my request was simply if it would be possible to automatically add that to the description.

If it is difficult to infer the default value (like I am using it) and it would be easier to use one of the methods you mention,  I think option 1 (textual default) would fit my use case the best.  That also sounds like the simplest option to implement.
